pymatgen/core/surface.py

```diff
@@ -165,204 +160,108 @@ def __init__(
-                if site in equi_sites:
-                    return idx
-            raise ValueError("Cannot determine equi index!")
+    def __str__(self) -> str:
+        outs = [
+            f"Slab Summary ({self.composition.formula})",
+            f"Reduced Formula: {self.composition.reduced_formula}",
+            f"Miller index: {self.miller_index}",
+            f"Shift: {self.shift:.4f}, Scale Factor: {self.scale_factor}",
+            f"abc   : {' '.join(f'{i:0.6f}'.rjust(10) for i in self.lattice.abc)}",
+            f"angles: {' '.join(f'{i:0.6f}'.rjust(10) for i in self.lattice.angles)}",
+            f"Sites ({len(self)})",
+        ]
 
-        for surface_site, shift in [(sorted_csites[0], slab_ratio), (sorted_csites[-1], -slab_ratio)]:
-            to_move = []
-            fixed = []
-            for site in sites:
-                if abs(site.c - surface_site.c) < tol and (
-                    (not same_species_only) or site.species == surface_site.species
-                ):
-                    to_move.append(site)
-                else:
-                    fixed.append(site)
+        for idx, site in enumerate(self):
+            outs.append(f"{idx + 1} {site.species_string} {' '.join(f'{j:0.6f}'.rjust(12) for j in site.frac_coords)}")
 
-            # Sort and group the sites by the species and symmetry equivalence
-            to_move = sorted(to_move, key=equi_index)
+        return "\n".join(outs)
 
-            grouped = [list(sites) for k, sites in itertools.groupby(to_move, key=equi_index)]
+    @property
+    def dipole(self) -> np.ndarray:
+        """Calculate the dipole moment of the Slab in the direction of the surface normal.
 
-            if len(to_move) == 0 or any(len(g) % 2 != 0 for g in grouped):
-                warnings.warn(
-                    "Odd number of sites to divide! Try changing "
-                    "the tolerance to ensure even division of "
-                    "sites or create supercells in a or b directions "
-                    "to allow for atoms to be moved!"
-                )
-                continue
-            combinations = []
-            for g in grouped:
-                combinations.append(list(itertools.combinations(g, int(len(g) / 2))))
+        Note that the Slab must be oxidation state decorated for this to work properly.
+        Otherwise, the Slab will always have a dipole moment of 0.
+        """
+        centroid = np.sum(self.cart_coords, axis=0) / len(self)
 
-            for selection in itertools.product(*combinations):
-                species = [site.species for site in fixed]
-                frac_coords = [site.frac_coords for site in fixed]
+        dipole = np.zeros(3)
+        for site in self:
+            charge = sum(getattr(sp, "oxi_state", 0) * amt for sp, amt in site.species.items())
+            dipole += charge * np.dot(site.coords - centroid, self.normal) * self.normal
```

Comment:
Need confirmation: The formula for calculating dipole moment seems not correct. Should be `charge * np.dot(site.coords - centroid, self.normal) / self.normal`?